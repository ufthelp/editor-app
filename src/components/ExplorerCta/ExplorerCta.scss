.explorerCta {
  cursor: pointer;
  border-left: 1px solid transparent;
  .childCta{
    padding: .5rem;
    display: flex;
    background-color: #333333;
    align-items: center;
    .text{
      text-transform: uppercase;
      font-size: .75rem;
      margin-left: .75rem;
    }
  }
}