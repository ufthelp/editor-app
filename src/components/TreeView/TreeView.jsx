import React, { Component } from "react";
import "./TreeView.scss";

class TreeView extends Component {
  state = {
    data: [
      {
        text: "Furniture",
        expanded: true,
        items: [
          { text: "Tables & Chairs" },
          { text: "Sofas" },
          { text: "Occasional Furniture" }
        ]
      },
      {
        text: "Decor",
        items: [
          { text: "Bed Linen" },
          { text: "Curtains & Blinds" },
          { text: "Carpets" }
        ]
      }
    ]
  };

  onExpandChange = node => {
    const { data } = this.state;
    if (!node.expanded) {
      node["expanded"] = false;
    }
    node.expanded = !node.expanded;

    const updatedNode = data.map(item => {
      if (item.text === node.text) {
        return node;
      }
      return item;
    });

    this.setState({
      data: updatedNode
    });
  };

  render() {
    const { data } = this.state;
    return (
      <>
        {data.map(node => (
          <TreeNode key={`node_${node.text}`}node={node} onExpandChange={this.onExpandChange} />
        ))}
      </>
    );
  }
}

const TreeNode = props => {
  const { node, onExpandChange } = props;
  return (
    <>
      <div onClick={() => onExpandChange(node)}>{node.text}</div>
      {node.expanded &&
        node.items.map((childNode, idx) => (
          <div className="childNode" key={`${childNode}_${idx}`}>
            {childNode.text}
          </div>
        ))}
    </>
  );
};
export default TreeView;
