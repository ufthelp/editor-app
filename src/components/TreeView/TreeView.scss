.childNode{
  margin-left: .75rem;
}