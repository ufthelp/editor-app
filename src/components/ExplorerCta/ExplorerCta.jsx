import React, { Component } from "react";
import TreeView from "../TreeView";

import "./ExplorerCta.scss";

class ExplorerCta extends Component {
  render(){
    return(
      <div className="explorerCta">
        <TreeView />
    </div>
    )
  }
}

export default ExplorerCta;
