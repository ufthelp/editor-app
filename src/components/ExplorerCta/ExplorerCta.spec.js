import React from 'react';
import {shallow} from 'enzyme';
import ExplorerCta from './ExplorerCta';
import TreeView from "../TreeView";


describe('<Explorer /> Component', ()=>{
    let wrapper;
    beforeEach(()=> wrapper = shallow( <ExplorerCta 
    />))

    it('should render correctly', () => expect(wrapper).toMatchSnapshot());

    it('should render  TreeView', () => {
        expect(wrapper.text()).toEqual('<TreeView />');
    });
})

