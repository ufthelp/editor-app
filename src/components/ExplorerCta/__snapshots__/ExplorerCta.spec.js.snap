// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Explorer /> Component should render correctly 1`] = `ShallowWrapper {}`;
